File: clients/csharp/sdk/Models/Registrations.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace MastodonClientLib.Models {
    public class Registrations : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>The approval_required property</summary>
        public bool? ApprovalRequired { get; set; }
        /// <summary>The enabled property</summary>
        public bool? Enabled { get; set; }
        /// <summary>The message property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Message { get; set; }
#nullable restore
#else
        public string Message { get; set; }
#endif
        /// <summary>
        /// Instantiates a new Registrations and sets the default values.
        /// </summary>
        public Registrations() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static Registrations CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new Registrations();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"approval_required", n => { ApprovalRequired = n.GetBoolValue(); } },
                {"enabled", n => { Enabled = n.GetBoolValue(); } },
                {"message", n => { Message = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteBoolValue("approval_required", ApprovalRequired);
            writer.WriteBoolValue("enabled", Enabled);
            writer.WriteStringValue("message", Message);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: clients/node/sdk/models/registrations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registrations = void 0;
class Registrations {
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Gets the approval_required property value. The approval_required property
     * @returns a boolean
     */
    get approval_required() {
        return this._approval_required;
    }
    ;
    /**
     * Sets the approval_required property value. The approval_required property
     * @param value Value to set for the approval_required property.
     */
    set approval_required(value) {
        this._approval_required = value;
    }
    ;
    /**
     * Instantiates a new Registrations and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * Gets the enabled property value. The enabled property
     * @returns a boolean
     */
    get enabled() {
        return this._enabled;
    }
    ;
    /**
     * Sets the enabled property value. The enabled property
     * @param value Value to set for the enabled property.
     */
    set enabled(value) {
        this._enabled = value;
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "approval_required": n => { this.approval_required = n.getBooleanValue(); },
            "enabled": n => { this.enabled = n.getBooleanValue(); },
            "message": n => { this.message = n.getStringValue(); },
        };
    }
    ;
    /**
     * Gets the message property value. The message property
     * @returns a string
     */
    get message() {
        return this._message;
    }
    ;
    /**
     * Sets the message property value. The message property
     * @param value Value to set for the message property.
     */
    set message(value) {
        this._message = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeBooleanValue("approval_required", this.approval_required);
        writer.writeBooleanValue("enabled", this.enabled);
        writer.writeStringValue("message", this.message);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
}
exports.Registrations = Registrations;
